### PR TITLE
Implement saveAclFor()

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

This adds the ability to save an ACL Resource back to the Pod, to associate it with the given Resource.

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated. (API docs are there, tutorial update will follow later.)
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: src/litDataset.test.ts [hunk 1]
```diff
@@ -1657,3 +1659,150 @@ describe("saveLitDatasetInContainer", () => {
...
+    });
+
+    const savedAcl = await unstable_saveAclFor(withResourceInfo, aclResource, {
+      fetch: mockFetch,
+    });
+
+    expect(savedAcl.accessTo).toBe("https://some.pod/resource");
+  });
+
+  it("sends a PATCH if the ACL contains a ChangeLog and was originally fetched from the same location", async () => {
```
Comment (NSeydoux) [inline src/litDataset.test.ts:1757]: I think this is a non-issue, but I'm asking just to be sure: if the ACL does not exist in the first place (its IRI as advertised by the resource Link header returns a 404), is its `fetchedFrom` property set to this IRI anyways ? And if so, a `PATCH` can create a resource, right ?

Comment (Vinnl) [reply src/litDataset.test.ts:1757]: I'm not sure if `PATCH` can create a Resource, but if it didn't exist, then this ACL will either have to have been created using `createLitDataset()` (in which case it will not have a `fetchedFrom`), or will have to have been based on a different Resource's ACL (in which case its `fetchedFrom` should be pointing to a different location), so it shouldn't be doing a `PATCH`.

File: src/litDataset.ts [hunk 2]
```diff
@@ -408,6 +409,38 @@ export async function saveLitDatasetInContainer(
...
 
+/**
+ * Save the ACL for a Resource.
+ *
+ * @param resource The Resource to which the given ACL applies.
+ * @param resourceAcl An [[unstable_AclDataset]] whose ACL Rules will apply to `resource`.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ */
+export async function unstable_saveAclFor(
+  resource: WithResourceInfo & {
```
Comment (ajacksified) [inline src/litDataset.ts:420]: This type is a little hard to follow - with unions and excludes, at what point does it just make sense to create a new type (even if this is the implementation underneath)? Could be my unfamiliarity with Typescript, but it _feels_ like a smell if your type definition is 7 lines long.

Comment (Vinnl) [reply src/litDataset.ts:420]: Yep, that's fair - in fact, I think I did want to create a separate type alias for that after I finished it, but forgot. I'll do it in a separate PR, since, you know... It involves new names :P

Comment (Vinnl) [reply src/litDataset.ts:420]: https://github.com/inrupt/lit-pod/pull/206
